constants.py

```diff
@@ -39,10 +41,12 @@ def _resource_path(relative_path: Path | str) -> Path:
 
     Works for dev and for PyInstaller.
     """
-    if IS_PACKAGED:
+    if IS_PACKAGED and not IS_APPIMAGE:
         # PyInstaller's folder where the one-file app is unpacked
         meipass: str = getattr(sys, "_MEIPASS")
         base_path = Path(meipass)
+    elif IS_APPIMAGE:
+        base_path = Path(sys.argv[0]).absolute().parent
```

Comment:
The `Path(sys.argv[0]).absolute()` part is just `SELF_PATH`, while the entire thing matches how `WORKING_DIR` is defined: `SELF_PATH.absolute().parent`. Further down the code, the `WORKING_DIR` is redefined as `Path(os.environ["APPIMAGE"]).absolute().parent`.

According to [AppImage docs](https://docs.appimage.org/packaging-guide/environment-variables.html), `APPIMAGE` env var points to the executable, which implies this has to be `True`:

`Path(sys.argv[0]) == Path(os.environ["APPIMAGE"])`

If that's always the case, then the `WORKING_DIR` redefinition (L85-L88) isn't needed, as it'd stay as `SELF_PATH.absolute().parent`. This modification at L48-L49 wouldn't be needed either, as L44 would already be sufficient to repoint `base_path` to `WORKING_DIR`, which would be the same folder per the explanation above.

Please test if applying these changes still results in a working application, and if so, I'd like them to become a part of this PR.

Reply:
Thanks. I'll get back to you on this.

Reply:
Just a small additional explanation note - a PyInstaller one-file application gets unpacked into a temporary folder somewhere in the Windows' `%TEMP%` region, where all of the additional data files are stored - these include the icon file, as well as the translation files. These are accessed via `_resource_path` usage, with the `sys._MEIPASS` variable being a path pointing to that temporary folder, as long as the application is packaged.

If it isn't packaged, then all translation files, the icon, etc., are available next to the executable Python code. Essentially, this makes the resource folder and `WORKING_DIR` be the same folder (when not packaged). When packaged, `WORKING_DIR` points to where the executable lives (cache, settings, etc.), while `_resource_path` points to where the temporarily unpacked data files live.

The data files are defined by this portion of code:
https://github.com/DevilXD/TwitchDropsMiner/blob/1e31957388869c56e2b21da38d11cc20dbde7cc2/build.spec#L19-L35
It's mostly just icon file, some seleniumwire certs from back when they were needed (not included now), and translation files.

I don't know how AppImage does it, but you can have either the resource folder and working directory be the same folder, or split those up into two like it's done with PyInstaller. Maybe that's what this `APPIMAGE` env va is for, and the paths can differ after all?

Reply:
>Maybe that's what this APPIMAGE env va is for, and the paths can differ after all?

That's precisely it, and I had forgotten about this.

`argv[0]` does not match `$APPIMAGE` because similarly to the PyInstaller executable, when you run the AppImage, all of its contents are mounted to a temporary directory (`/tmp` on my machine), and only then the `main.py` script is invoked.

And as far as I know, AppImages do not offer a mechanism like PyInstaller's `_resource_path`.

So, at least in the AppImage's case, `argv[0]` points to `/tmp/.mount_Twitch<random string>/usr/src/main.py`, while `$APPIMAGE` actually points to `/home/gui/TDM/Twitch.Drops.Miner-x86_64.AppImage`, which is what we want.

So I think that the concept behind `SELF_PATH` and `WORKING_DIR` don't apply really well in AppImage's case and can be a bit confusing, so with my last push I introduced a new variable (`APPIMAGE_PATH`) that should hopefully make things a bit clearer and less redundant.

In any case, please share your thoughts.